internal/agent/config.go

```diff
@@ -66,13 +104,68 @@ func (cfg *Config) SetEnrollmentMetricsCallback(cb func(operation string, duract
 	cfg.enrollmentMetricsCallback = cb
 }
 
-// TODO: dedupe with internal/config/config.go
+// Validate checks that the required fields are set and that the paths exist.
+func (cfg *Config) Validate() error {
+	requiredFields := []struct {
+		value     string
+		name      string
+		checkPath bool
+	}{
+		{cfg.ManagementEndpoint, "management-endpoint", false},
+		{cfg.EnrollmentEndpoint, "enrollment-endpoint", false},
+		{cfg.GeneratedCert, "generated-cert", false},
+		{cfg.ConfigDir, "config-dir", true},
+		{cfg.DataDir, "data-dir", true},
```

Comment:
I would say that we could create data-dir if it does not exist. We may need to do it in .spec %post install otherwise, because /var/lib does not get from the rpm package for ostree... may be there is some inner magic to handle it though.